File: 4057-TotalWavinessOfNumbersInRangeI/4057-TotalWavinessOfNumbersInRangeI.js
// Last updated: 20/07/2026, 11:23:40
var totalWaviness = function(num1, num2) {
    let ans = 0;
    for (let num = num1; num <= num2; num++) {
        const s = num.toString();
        for (let i = 1; i < s.length - 1; i++) {
            if ((s[i] > s[i - 1] && s[i] > s[i + 1]) ||
                (s[i] < s[i - 1] && s[i] < s[i + 1])) {
                ans++;
            }
        }
    }
    return ans;
};

File: 4057-TotalWavinessOfNumbersInRangeI/4057-TotalWavinessOfNumbersInRangeI.java
// Last updated: 20/07/2026, 11:23:41
class Solution {
    public int totalWaviness(int num1, int num2) {
        int ans = 0;
        for (int i = num1; i <= num2; i++) {
            String s = String.valueOf(i);
            for (int j = 1; j < s.length() - 1; j++) {
                if ((s.charAt(j) > s.charAt(j - 1) && s.charAt(j) > s.charAt(j + 1)) ||
                    (s.charAt(j) < s.charAt(j - 1) && s.charAt(j) < s.charAt(j + 1))) {
                    ans++;
                }
            }
        }
        return ans;
    }
}